titus-client/src/main/java/com/netflix/titus/runtime/endpoint/v3/grpc/GrpcJobManagementModelConverters.java

```diff
@@ -1071,6 +1102,32 @@ public static Capacity toGrpcCapacity(com.netflix.titus.api.jobmanager.model.job
                 .build();
     }
 
+    private static List<com.netflix.titus.grpc.protogen.PlatformSidecar> toGrpcPlatformSidecars(List<PlatformSidecar> platformSidecarList) {
+        List<com.netflix.titus.grpc.protogen.PlatformSidecar> platformSidecars = new ArrayList<>();
+        for (PlatformSidecar ps : platformSidecarList) {
+            platformSidecars.add(toGrpcPlatformSidecar(ps));
+        }
+        return platformSidecars;
+    }
+
+    private static com.netflix.titus.grpc.protogen.PlatformSidecar toGrpcPlatformSidecar(PlatformSidecar ps) {
+        return com.netflix.titus.grpc.protogen.PlatformSidecar.newBuilder()
+                .setName(ps.getName())
+                .setChannel(ps.getChannel())
+                .setArguments(jsonStringToStruct(ps.getArguments(), ps.getName()))
+                .build();
+    }
+
+    private static Struct jsonStringToStruct(String arguments, String sidecarName) {
+        Struct.Builder argsBuilder = Struct.newBuilder();
+        try {
+            JsonFormat.parser().merge(arguments, argsBuilder);
+        } catch (InvalidProtocolBufferException e) {
+            throw new IllegalArgumentException("Could not serialize platform sidecar arguments for " + sidecarName + ": " + e.getMessage());
```

Comment:
This will prevent any external access to this job. I think we should return here `Optional<String>`, and log information about mapping error.

Reply:
I see what you mean. 
I hope that never happens, it would be a little hard to debug.
I hope logging that error yet still returning empty Struct isn't going to have any other weird repercussions I can't think about...

Reply:
Can you confirm, you would like this method to return `Pair<Struct, Optional<String>>` and have the caller deal with the String if present? (kinda golang style)

And in the `toGrpcPlatformSidecar` method, if it sees that optional string, just log instead?

Reply:
Yes, that would be even better. You can use `Either` type for that (from titus-common).

Reply:
Used `Either`